### PR TITLE
Add new CLI commands to dataset factory docs

### PR DESCRIPTION
**Pull request by AhdraMeraliQB**
> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.
## Description
Resolves https://github.com/kedro-org/kedro/issues/2849

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

### CODE REVIEW

File: docs/source/data/kedro_dataset_factories.md [hunk 1]
```diff
@@ -225,3 +225,145 @@ You can use dataset factories to define a catch-all pattern which will overwrite
...
 Kedro will now treat all the datasets mentioned in your project's pipelines that do not appear as specific patterns or explicit entries in your catalog
 as `pandas.CSVDataSet`.
+
+## CLI commands for dataset factories
+
+To manage your dataset factories, two new commands have been added to the Kedro CLI: `kedro catalog rank` (0.18.12) and `kedro catalog resolve` (0.18.13).
+
+#### Using `kedro catalog rank`
+
+This command outputs a list of all dataset factories in the catalog, ranked in the order by which pipeline datasets are matched against them. This ordering is determined by considering the following criteria:
```
Comment (stichbury) [inline docs/source/data/kedro_dataset_factories.md:235]:
```suggestion
This command outputs a list of all dataset factories in the catalog, ranked in the order by which pipeline datasets are matched against them. The ordering is determined by the following criteria:
```

File: docs/source/data/kedro_dataset_factories.md [hunk 2]
```diff
@@ -225,3 +225,145 @@ You can use dataset factories to define a catch-all pattern which will overwrite
...
+- processed_{dataset_name}
+- '{namespace}.{dataset_name}_pq'
+- '{dataset_name}_csv'
+- '{layer}.{dataset_name}'
+- '{default_dataset}'
+```
+
+As we can see, the entries are ranked firstly by how many non-placeholders are in the pattern, in descending order. Where two entries have the same number of non-placeholder characters, `{namespace}.{dataset_name}_pq` and `{dataset_name}_csv` with four each, they are then ranked by the number of placeholders, also in decreasing order. `{default_dataset}` is the least specific pattern possible, and will always be matched against last.
+
+#### Using `kedro catalog resolve`
```
Comment (stichbury) [inline docs/source/data/kedro_dataset_factories.md:282]:
```suggestion
#### How to use `kedro catalog resolve`
```

File: docs/source/data/kedro_dataset_factories.md [hunk 3]
```diff
@@ -225,3 +225,145 @@ You can use dataset factories to define a catch-all pattern which will overwrite
...
+
+This command outputs a list of all dataset factories in the catalog, ranked in the order by which pipeline datasets are matched against them. This ordering is determined by considering the following criteria:
+
+1. The number of non-placeholder characters in the pattern
+2. The number of placeholders in the pattern
+3. Alphabetic ordering
+
+Consider a catalog file with the following patterns:
+
+```yaml
```
Comment (stichbury) [inline docs/source/data/kedro_dataset_factories.md:243]: I'm leaning towards suggesting that we put the code blocks in the click to open <summary><details> tags. WDYT?

**Review by stichbury — APPROVED** [review]
This looks good to me. Just a couple of minor suggestions, nothing blocking so I've approved. Thank you @AhdraMeraliQB 🌟

**Review by stichbury — APPROVED** [review]
This looks good to me. Just a couple of minor suggestions, nothing blocking so I've approved. Thank you @AhdraMeraliQB 🌟

**Review by stichbury — APPROVED** [review]
This looks good to me. Just a couple of minor suggestions, nothing blocking so I've approved. Thank you @AhdraMeraliQB 🌟

File: docs/source/data/kedro_dataset_factories.md [hunk 4]
```diff
@@ -225,3 +225,145 @@ You can use dataset factories to define a catch-all pattern which will overwrite
...
+- preprocessed_{dataset_name}
+- processed_{dataset_name}
+- '{namespace}.{dataset_name}_pq'
+- '{dataset_name}_csv'
+- '{layer}.{dataset_name}'
+- '{default_dataset}'
+```
+
+As we can see, the entries are ranked firstly by how many non-placeholders are in the pattern, in descending order. Where two entries have the same number of non-placeholder characters, `{namespace}.{dataset_name}_pq` and `{dataset_name}_csv` with four each, they are then ranked by the number of placeholders, also in decreasing order. `{default_dataset}` is the least specific pattern possible, and will always be matched against last.
+
```
Comment (merelcht) [inline docs/source/data/kedro_dataset_factories.md:281]: This is great! 👍

**Review by merelcht — APPROVED** [review]
LGTM! 👍

File: docs/source/data/kedro_dataset_factories.md (hunk 3 above)
Comment (AhdraMeraliQB) [reply docs/source/data/kedro_dataset_factories.md:246]: Sounds good, changed in [507c1d1](https://github.com/kedro-org/kedro/pull/2935/commits/507c1d142a03ef7c667d1b43b8eb39a853dd496e). I left the output of `kedro catalog rank` as it was as it's rather short and isn't technically a code snippet, but happy to make it uniform too

**Review by ankatiyar — APPROVED** [review]
👍🏾

**Pull request merged**